classes/Manufacturer.php

```diff
@@ -381,7 +381,7 @@ public static function getProducts(
         }
 
         $front = true;
-        if (!in_array($context->controller->controller_type, ['front', 'modulefront'])) {
+        if (isset($context->controller) && !in_array($context->controller->controller_type, ['front', 'modulefront'])) {
```

Comment:
```suggestion
        if (!isset($context->controller) || !in_array($context->controller->controller_type, ['front', 'modulefront'])) {
```
If the controller is not accessible it is very likely because it's a context that is not the front no?

Reply:
Yes but the current implementation tends to fallback more easily on the default `$front = true;` But the absence of a controller to check is more likely to happen in a context where we are NOT on the front

The Admin API is a good example since it is what motivated this fix, I may be wrong but with the current fix in admin API where we don't have a legacy controller accessible the Manufacturer will "think" it is in FO

That's why I suggested the inverse condition when no controller is detected we should have `$front = false;` hence the above suggestion